index.html

```diff
@@ -164,14 +184,34 @@ <h2>Becoming a full member of a section</h2>
         JoiningRelocateCandidate["JoiningRelocateCandidate<br />(Take RelocatedInfo)"]
         style JoiningRelocateCandidate fill:#f9f,stroke:#333,stroke-width:4px
 
-        InitialSendConnectionInfoRequest["RELOCATED_INFO = RelocatedInfo<br/><br/>For all elders in RelocatedInfo.section_info:<br/>send_rpc(ConnectionInfoRequest)<br/><br/>schedule(TimeoutRefused)"]
+        InitialSendConnectionInfoRequest["RELOCATED_INFO = RelocatedInfo<br/>
+        <br/>
+        For node in RELOCATED_INFO.section_info:<br/>
+        send_rpc(ConnectionInfoRequest) to node<br/>
+        <br/>
+        schedule(TimeoutRefused)<br/>
+        schedule(TimeoutResendInfo)"]
         InitialSendConnectionInfoRequest-->LoopStart
 
         LoopStart --> WaitFor
         WaitFor(("Wait for 0:"))
         LocalEvent((Local<br />Event))
         WaitFor --> LocalEvent
 
+        LocalEvent--"TimeoutResendInfo triggered"--> CheckResend
+        CheckResend["(connected, unconnected) =<br/>get_connected_and_unconnected(<br/>RELOCATED_INFO.section_info)<br/>
+        <br/>
+        For node in unconnected:<br/>
+        send_rpc(ConnectionInfoRequest) to node<br/>
+        <br/>
+        For node in connected:<br/>
+        if !has_resource_proof_for(node):<br/>
+        send_rpc(<br/>Rpc::CandidateInfo from RELOCATED_INFO)<br/>to node<br/>
```

Comment:
Shouldn't there be a mechanism for not always resending the CandidateInfo once they've received it?

Reply:
Maybe that can be fixed by making `get_connected_and_unconnected ` return only a subset of the initial nodes and ignoring any node for which we received a `ResourceProof` RPC. That would need to be explicit, though. Also, maybe the name would then be  a bit misleading if unchanged.

Reply:
> Shouldn't there be a mechanism for not always resending the CandidateInfo once they've received it?

Reply:
> Shouldn't there be a mechanism for not always resending the CandidateInfo once they've received it?

There is, it is when we received a ResourceProof Rpc from the elder, which is when we know they received the CandidateInfo

Reply:
If you are talking about connected node, I don't see why we should special case the ones were we have received ResourceProof, if we get disconnected we reconnect. Do you see a problem with that?

Reply:
OK. I see. I had missed the `if !has_resource_proof` condition in the diagram. It's more or less what I was advocating for. Looks good :smile: